### DISCUSSION TITLE
Keeping wildcard characters for Dockerfile package updates

### DISCUSSION THREAD

**Discussion by adamadamos** [renovatebot/renovate#30320] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Renovate seems to replace wildcard characters when updating packages in Dockerfiles. For example, when installing the GitHub CLI. 
```dockerfile
# renovate: datasource=github-releases depName=gh packageName=cli/cli
ENV GH_VERSION=v1.*
```
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "customManagers:dockerfileVersions"
  ]
}
```
**Expected**: `v1.*` to `v2.*`
**Actual**: `v1.*` to `v2.53.0`

Tried a few different versioning options including regex, [`^(?<major>\d+.\*)$`](https://regex101.com/r/tojjF1/1), but didn't have much luck. Just curious if anyone has come across something similar before, and if this is expected behaviour or not.  

Minimal reproduction: https://github.com/adamadamos/renovate-docker-wildcard-reproduction

Thanks

### Logs (if relevant)

<details><summary>Logs</summary>

```json
{
"depName": "gh",
"packageName": "cli/cli",
"currentValue": "v1.*",
"datasource": "github-releases",
"replaceString": "# renovate: datasource=github-releases depName=gh packageName=cli/cli\nENV GH_VERSION=v1.*\n",
"updates": [
  {
    "bucket": "major",
    "newVersion": "v2.53.0",
    "newValue": "v2.53.0",
    "releaseTimestamp": "2024-07-17T18:51:32.000Z",
    "newMajor": 2,
    "newMinor": 53,
    "newPatch": 0,
    "updateType": "major",
    "branchName": "renovate/major-dockerfile-dependencies"
  }
],
"versioning": "semver-coerced",
"warnings": [],
"sourceUrl": "https://github.com/cli/cli",
"registryUrl": "https://github.com",
"currentVersion": "v1.14.0",
"currentVersionTimestamp": "2021-08-04T14:20:44.000Z",
"isSingleVersion": true,
"fixedVersion": "v1.*"

```

</details>

**Comment by rarkins** [comment]
You need to experiment with a different Renovate `versioning` which understands such syntax, e.g. `semver` might solve it. I suspect it's using `semver-coerced` currently

**Reply by adamadamos** [reply, marked as answer]
Thanks, versioning with [npm](https://docs.renovatebot.com/modules/versioning/npm/) seems to do the trick. 
```dockerfile
# renovate: datasource=github-releases depName=gh packageName=cli/cli versioning=npm
ENV GH_VERSION=v1.*
```
